### PR TITLE
feat: add search source suggestions option

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Add possibility to search for sources
- API changes: https://github.com/dailydotdev/daily-api/pull/1985

Screenshot 2024-06-12 at 15 20 28

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-371 #done 

### Preview domain
https://as-371-search-sources.preview.app.daily.dev

### CODE REVIEW

**Review by nensidosari — APPROVED** [review]
Looks great!

File: packages/shared/src/components/search/SearchPanel/SearchPanelSourceSuggestions.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+const PanelItem = ({ suggestion, ...rest }: PanelItemProps) => {
+  const Icon = () => (
+    <img
+      loading="lazy"
+      src={suggestion.image}
+      alt={`${suggestion.title} logo`}
+      className="size-7 rounded-full"
+    />
```
Comment (capJavert) [inline packages/shared/src/components/search/SearchPanel/SearchPanelSourceSuggestions.tsx:30]: Use our `Image` component?

**Review by capJavert — APPROVED** [review]
Nice work!

File: packages/shared/src/components/search/SearchPanel/SearchPanelSourceSuggestions.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/search/SearchPanel/SearchPanelSourceSuggestions.tsx:30]: Ill check why i opted to use this in source page, feel it had some reason but dont remember 😂

**Pull request merged**